### PR TITLE
Wikipedia-Style Tooltips

### PR DESCRIPTION
**Pull request by stkeogie**
Implemented Tooltips (Wikipedia-Style Component) #289 
Here is a small preview of the implemented features.
tooltip1

### CODE REVIEW

File: app/src/app/shared/components/tooltip/tooltip.component.scss [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  object-fit: cover;
+  max-width: 100%;
+  max-height: 100%;
+  width: 100%;
+}
+.show-tooltip{
+  visibility: visible;
+}
+
+.right:before {
```
Comment (EeveesEyes) [inline app/src/app/shared/components/tooltip/tooltip.component.scss:53]: such classes can be included in the parent class using `&:before`. Same applies to `.tooltip { &-text {} &-image {} }` and so on.

Comment (stkeogie) [reply app/src/app/shared/components/tooltip/tooltip.component.scss]: Thank you for your feedback. I moved the tooltip-component code to the badge-component and removed the tooltip component.

File: app/src/app/shared/components/badge/badge.component.html [hunk 2]
```diff
@@ -12,3 +12,15 @@
...
+
+
+<ng-template #htmlContent>
+  <div class="tooltip-container">
+    <div class="tooltip-text-container">
+      <span  class="tooltip-text">
+        {{ tooltip }}
+      </span>
+    </div>
+    <div *ngIf="image" class="tooltip-image-container"><img id="tooltip-image" [src]="image" /></div>
```
Comment (EeveesEyes) [inline app/src/app/shared/components/badge/badge.component.html:24]: please use entity.imageSmall here. No image input needed.

File: app/src/app/shared/components/badge/badge.component.scss [hunk 3]
```diff
@@ -0,0 +1,26 @@
+
+.tooltip-container {
+    display: flex;
+    flex-direction: row;
+    height: 100%;
+}
+.tooltip-text-container {
+  padding: 16px;
+  flex: 1 1 0px;
+  word-wrap: anywhere;
```
Comment (EeveesEyes) [inline app/src/app/shared/components/badge/badge.component.scss:10]: I think this doesn't work. Maybe word-wrap works different.

File: app/src/app/shared/components/badge/badge.component.ts [hunk 4]
```diff
@@ -12,6 +12,7 @@ export class BadgeComponent implements OnInit, OnChanges {
   @Input() entity: Entity;
   @Input() isHoverBadge: boolean;
   @Input() hoveredArtwork: Artwork;
+  @Input() image: string;
```
Comment (EeveesEyes) [inline app/src/app/shared/components/badge/badge.component.ts:15]: See above :)

**Comment by stkeogie** [comment]
@EeveesEyes 
I fixed the inverted arrow bug, but It looks like the bug existed before my changes so the probability is high that the bug is currently live.
Regarding the max-height for the tooltip-image, I don't Think that we need a max-height. The image is automatically scaled down if its to large.

**Comment by jschurse** [comment]
@EeveesEyes has this been resolved?

**Pull request merged**